homeassistant/components/volvooncall/__init__.py

```diff
@@ -128,76 +129,48 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         region=config[DOMAIN].get(CONF_REGION),
     )
 
-    interval = config[DOMAIN][CONF_SCAN_INTERVAL]
+    hass.data[DATA_KEY] = {}
 
-    data = hass.data[DATA_KEY] = VolvoData(config)
+    coordinator = VolvoUpdateCoordinator(hass)
 
-    def is_enabled(attr):
-        """Return true if the user has enabled the resource."""
-        return attr in config[DOMAIN].get(CONF_RESOURCES, [attr])
+    data = hass.data[DATA_KEY] = VolvoData(hass, connection, config, coordinator)
```

Comment:
Pass the VolvoaData instance to the coordinator instead of the other way around. Store the coordinator in hass.data.